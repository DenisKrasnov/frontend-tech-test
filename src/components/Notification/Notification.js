import React from "react";
import { oneOf, func, node } from "prop-types";
import cn from "classnames";

import Icon from "../Icon";
import styles from "./Notification.scss";

const Notification = props => {
  const { type, onDismiss, children } = props;

  const className = cn(styles.Notification, styles[type]);

  return (
    <div className={className}>
      {children}
      {onDismiss ? (
        <div className={styles.dismiss}>
          <Icon color="white" glyph="x" onClick={onDismiss} />
        </div>
      ) : null}
    </div>
  );
};

Notification.defaultProps = {
  onDismiss: () => {},
  children: <span />,
};

Notification.propTypes = {
  type: oneOf(["success", "error"]).isRequired,
  onDismiss: func,
  children: node,
};

export default Notification;
