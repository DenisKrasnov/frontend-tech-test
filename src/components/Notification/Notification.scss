.Notification {
  display: flex;
  align-self: center;
  justify-content: center;
  min-width: 50%;
  padding: 16px 22px 14px;
  margin-bottom: 15px;
  border-radius: 4px;
  color: #fff;
  line-height: 18px;
  transition: width 3s linear;
}

.success {
  background: $success-color;
}

.error {
  background: $error-color;
}

.dismiss {
  margin: 0 0 0 16px;
  text-align: center;
}
