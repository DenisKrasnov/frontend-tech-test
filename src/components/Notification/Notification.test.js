import React from "react";
import { shallow } from "enzyme";

import Notification from "./Notification";

describe("Notification", () => {
  it("renders", () => {
    const component = shallow(<Notification type="success" />);

    expect(component).toMatchSnapshot();
  });

  it("renders with children", () => {
    const component = shallow(
      <Notification type="success">
        <span>Some Text</span>
      </Notification>
    );

    expect(component).toMatchSnapshot();
  });

  it("renders with onDismiss Icon and triggers the onDismiss", () => {
    const onDismiss = () => {};
    const component = shallow(
      <Notification onDismiss={onDismiss} type="success">
        <span>Some Text</span>
      </Notification>
    );

    expect(component).toMatchSnapshot();
  });
});
