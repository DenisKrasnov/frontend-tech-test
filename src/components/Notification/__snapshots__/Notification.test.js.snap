// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Notification renders 1`] = `
<div
  className="Notification success"
>
  <span />
  <div
    className="dismiss"
  >
    <Icon
      color="white"
      glyph="x"
      onClick={[Function]}
    />
  </div>
</div>
`;

exports[`Notification renders with children 1`] = `
<div
  className="Notification success"
>
  <span>
    Some Text
  </span>
  <div
    className="dismiss"
  >
    <Icon
      color="white"
      glyph="x"
      onClick={[Function]}
    />
  </div>
</div>
`;

exports[`Notification renders with onDismiss Icon and triggers the onDismiss 1`] = `
<div
  className="Notification success"
>
  <span>
    Some Text
  </span>
  <div
    className="dismiss"
  >
    <Icon
      color="white"
      glyph="x"
      onClick={[Function]}
    />
  </div>
</div>
`;
